src/libs/actions/App.js

```diff
@@ -184,9 +185,14 @@ function openApp() {
  * Refreshes data when the app reconnects
  */
 function reconnectApp() {
+    // When the app reconnects we do a fast "sync" of the LHN and only return chats that have new messages. We achieve this by sending the most recent reportActionID
```

Comment:
We briefly mentioned this before, but should we do this in OpenApp as well? Probably, yes?

Reply:
Yeah. I spent some time trying to figure this out and decided it is not so simple... 

Basically, we would need to change this code here:

https://github.com/Expensify/App/blob/ec3c9a268ffc00b2beb88eb206ea72d0a241878f/src/libs/Navigation/AppNavigator/AuthScreens.js#L122

And give it some knowledge of whether or not it should call `reconnectApp()` instead. I presumed maybe we can check if the user has "signed in" during that session, or whether they have existing report data, or something like that.

Both of those ideas ended up not really having a clear or obvious implementation.

Reply:
Ok, so, I figured out some way to do this. Will need to test the solution some more.

Reply:
Heh ok, so I got some code to detect if the user is signing in hooked up. But the changes from this [PR here](https://github.com/Expensify/App/pull/16093) is now cancelling the requests to `ReconnectApp` when the app opens 🤣 

I [side eyed](https://github.com/Expensify/App/pull/16093#pullrequestreview-1424890391) this PR last month so maybe we have a good excuse to revert it...

Reply:
Thanks ❤️ Another alternative is what we are discussing internally here: https://expensify.slack.com/archives/C05CBC62HGW/p1687184679739399?thread_ts=1686757873.460469&cid=C05CBC62HGW